### PR TITLE
Project

### PR DESCRIPTION
**Pull request by petr-shumskiy**

### CODE REVIEW

File: components/Message.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  return (
+    <li
+      style={{
+        listStyle: 'none',
+        textAlign: messageType === 'info' ? 'center' : 'start',
+        fontSize: '20px'
+      }}
+    >
+      {
+        user ? (
```
Comment (Arteezy19) [inline components/Message.tsx]: user && (
          <Typography
            component='span'
            style={{ marginRight: '8px', color }}
          >
            {user}
          </Typography>
        )

File: pages/greet.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  inputText: {
+    width: '70%',
+    textAlign: 'center',
+    'borderBottom': '2px solid #fff'
+  },
+  inputUnderline: {
+  }
+})
+
+export default function Greet(props: IProps) {
```
Comment (Arteezy19) [inline pages/greet.tsx]: IProps -> IGreet, interface should have component Name

File: pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  const onSubmit = (event: SyntheticEvent) => {
+    event.preventDefault()
+
+    setLogged(true)
+    setMessages((prevMessages) => [...prevMessages,])
+
+    socket.emit('add user', userName)
+  }
+
```
Comment (Arteezy19) [inline pages/index.tsx:40]: Too much useless /n

File: pages/index.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  const socket = io('http://192.168.0.144:3000').connect()
+  const greetingMessage = newMessageBuilder('info', null, 'Welcome to chat')
+
+  const [isLogged, setLogged] = useState(false)
+  const [messages, setMessages] = useState<IMessage[]>([greetingMessage])
+  const [userName, setUserName] = useState('')
+  const [colorMessage, setColor] = useState<ColorsType | null>(null)
+
+
+  useEffect(() => {
```
Comment (Arteezy19) [inline pages/index.tsx:21]: I'm not sure if we should use socket.on inside useEffect

File: pages/index.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import { SyntheticEvent, useEffect, useState } from 'react'
```
Comment (Arteezy19) [inline pages/index.tsx:1]: First of all should go downloaded packages, then custom packages, then styles
```
import { SyntheticEvent, useEffect, useState } from 'react'
import { io } from 'socket.io-client'

import Greet from './greet';
import { newMessageBuilder, getRandomColor } from '../utls';
import { IMessage, ColorsType, IServerUserJoinedOrLeft } from '../types';
import { Chat } from '../components/Chat';
```

File: pages/index.tsx (hunk 3 above)
Comment (petr-shumskiy) [reply pages/index.tsx:40]: removing of it breaks the functionality of info message about joining chat of anyone

**Pull request closed without merging**